subprojects/core-api/src/main/java/org/gradle/internal/deprecation/DeprecatableConfiguration.java

```diff
@@ -40,78 +38,85 @@
+     */
+    default void maybeEmitConsumptionDeprecation() {
+        if (isDeprecatedForConsumption()) {
+            DeprecationLogger.deprecateConfiguration(getName())
                 .forConsumption()
                 .willBecomeAnErrorInGradle9()
-                .withUserManual("dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations"));
+                .withUserManual("dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations")
+                .nagUser();
+        }
     }
 
     /**
-     * Allows plugins to deprecate the resolvability property (canBeResolved() == true) of a configuration that will be changed in the next major Gradle version.
-     *
-     * @param alternativesForResolving alternative configurations that can be used for dependency resolution
-     * @return this configuration
+     * If this configuration is deprecated for declaration, emit a deprecation warning.
      */
-    DeprecatableConfiguration deprecateForResolution(String... alternativesForResolving);
+    default void maybeEmitDeclarationDeprecation() {
+        if (isDeprecatedForDeclarationAgainst()) {
+            DeprecationLogger.deprecateConfiguration(getName())
+                .forDependencyDeclaration()
+                .replaceWith(getDeclarationAlternatives())
+                .willBecomeAnErrorInGradle9()
+                .withUpgradeGuideSection(5, "dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations")
+                .nagUser();
+        }
+    }
 
     /**
-     * Allows plugins to deprecate a configuration that will be removed in the next major Gradle version.
-     *
-     * @param alternativesForDeclaring alternative configurations that can be used to declare dependencies
-     * @return this configuration
+     * If this configuration is deprecated for resolution, emit a deprecation warning.
      */
-    DeprecatableConfiguration deprecateForDeclarationAgainst(String... alternativesForDeclaring);
+    default void maybeEmitResolutionDeprecation() {
+        if (isDeprecatedForResolution()) {
+            DeprecationLogger.deprecateConfiguration(getName()
+                ).forResolution()
+                .replaceWith(getResolutionAlternatives())
+                .willBecomeAnErrorInGradle9()
+                .withUpgradeGuideSection(5, "dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations")
+                .nagUser();
+        }
+    }
 
     boolean isDeprecatedForConsumption();
     boolean isDeprecatedForResolution();
     boolean isDeprecatedForDeclarationAgainst();
 
     default boolean canSafelyBeResolved() {
-        if (!isCanBeResolved()) {
-            return false;
-        }
-        List<String> resolutionAlternatives = getResolutionAlternatives();
-        return resolutionAlternatives == null;
+        return isCanBeResolved() && !isDeprecatedForResolution();
```

Comment:
❗ Not introduced in this PR, but this method is called to write dependency verification metadata.  Configurations that are resolvable but deprecated for resolution _can still be resolved_.  This could be a problem, since the metatdata won't be written for those confs.  I think this method should go away and we should just call `isCanBeResolved()` directly.

Reply:
If we didn't check for the deprecation then `--write-verification-metadata` would print deprecation warnings. I'm not sure this is something we want since users can't really do anything about that.

Reply:
I suppose since this is how it's been working there's no need to change anything now.  It just seems like an interesting path to avoiding recording metadata while still doing resolution.